src/main/java/com/autotune/analyzer/recommendations/engine/DurationBasedRecommendationEngine.java

```diff
@@ -92,61 +93,90 @@ public HashMap<String, Recommendation> generateRecommendation(ContainerData cont
                 HashMap<AnalyzerConstants.ResourceSetting, HashMap<AnalyzerConstants.RecommendationItem, RecommendationConfigItem>> config = new HashMap<>();
                 // Create Request Map
                 HashMap<AnalyzerConstants.RecommendationItem, RecommendationConfigItem> requestsMap = new HashMap<>();
+                // Recommendation Item checks
+                boolean isCpuRequestValid = true;
+                boolean isMemoryRequestValid = true;
                 // Get the Recommendation Items
-                RecommendationConfigItem cpuRequestItem = getCPURequestRecommendation(filteredResultsMap);
-                RecommendationConfigItem memRequestItem = getMemoryRequestRecommendation(filteredResultsMap);
+                RecommendationConfigItem cpuRequestItem = getCPURequestRecommendation(filteredResultsMap, monitoringEndTime);
+                RecommendationConfigItem memRequestItem = getMemoryRequestRecommendation(filteredResultsMap, monitoringEndTime);
+
+                if (cpuRequestItem.getAmount() <= 0) { isCpuRequestValid = false; }
+                if (memRequestItem.getAmount() <= 0) { isMemoryRequestValid = false; }
+
                 // Initiate generated value holders with min values constants to compare later
                 Double generatedCpuRequest = null;
                 String generatedCpuRequestFormat = null;
                 Double generatedMemRequest = null;
                 String generatedMemRequestFormat = null;
 
                 // Check for null
-                if (null != cpuRequestItem) {
+                if (null != cpuRequestItem && isCpuRequestValid) {
                     generatedCpuRequest = cpuRequestItem.getAmount();
                     generatedCpuRequestFormat = cpuRequestItem.getFormat();
                     requestsMap.put(AnalyzerConstants.RecommendationItem.cpu, cpuRequestItem);
+                } else {
+                    RecommendationNotification recommendationNotification = new RecommendationNotification(
+                            AnalyzerConstants.RecommendationNotification.CPU_RECORDS_ARE_ZERO
+                    );
+                    recommendation.addNotification(recommendationNotification);
```

Comment:
This should happen in getCPURequestRecommendation method itself. So we might need to send the `recommendationNotification` object as a parameter. This will help catch other scenarios which can only be caught inside the method.

Reply:
@dinogun Thanks for the review, I really appreciate your idea of passing notification to the function as param, But  after going through the code I have came up with points to be discussed before I could make some changes as suggested, Please correct me if I'm wrong.

- I'm more inclined keep the math calculation away from making structural changes to a data structure (Notification Object) whose scope is at a higher level
- As per your suggestion we can implement the notification updates at the calculation phase itself by implementing something like this

Generate recommendation calls `processCPUrecommendation` which returns a object with `requests` and `limits` map,  and a notification map

 `processCPUrecommendation` calls `checkCPURequest` (basically we call getcpurequest and based on output we set enum) which returns  enum which decribes the status (tells about issue clearly or states that there is no issue) and the value of cpurequest, if there is no issue we read the cpurequest value and format the `requests` map and if the enum shows issue we add the notification to map and return them to `generate recomemndation` and it reads the structures as it is and populate the recommendation object
 
 
 Please let me know your views on this

Reply:
I'd prefer not to do any processing of the data at the `generateRecommendation` level, instead they all need to happen inside `getXYZRequestRecommendation`